backend/src/databasefunctions.ts

```diff
@@ -0,0 +1,84 @@
+import { PoolConfig, ClientConfig } from 'pg';
+import { ErrorHandler } from './helpers/error';
+const { Pool } = require('pg');
+require('dotenv').config();
+
+const pgconfig: PoolConfig = {
+  user: process.env.DBUSER,
+  database: process.env.DATABASENAME,
+  password: process.env.DBPASSWORD,
+  host: process.env.DBHOST,
+  port: parseInt(process.env.DBPORT),
+};
+
+const pool = new Pool(pgconfig);
+
+pool.on('error', async (err: Error, client: ClientConfig) => {
+  throw new ErrorHandler(500, String(err), client);
+});
+
+const validSQLStatment = async (
+  type: string,
+  sqlStatement: string,
+) => {
+  if (!sqlStatement.includes(type)) {
+    throw new ErrorHandler(500, 'Invalid SQL statement');
+  }
+};
```

Comment:
Some minor preference-comments (do with them as you wish):

- rename the function to `validateSQLStatement`. Then is on the same form as `executeQuery`. Also, it is kind of more correct since in the function we are validating the statement. `valid` may mislead a bit indicating that it is a sql statement.
- change `type` to something less ambiguous, maybe `keyword` or `sqlKeyword`.
- move the expression in the if-statement to it's own function i.e. `const isValidSqlStatement...` or even better `containesCorrectKeyword` which returns the evaluation of the expression. Makes it easier to expand what a valid sql statement is later and generally makes thinks more readable.

Reply:
I agree, there's no reason to not be thorough.